src/conditions/context/context.ts

```diff
@@ -49,7 +50,10 @@ export class ConditionContext {
       }
     });
 
-    if (this.requiresSigner() && !this.signer) {
+    const conditionRequiresSigner = this.conditions.some((c) =>
+      c.requiresSigner()
+    );
+    if (conditionRequiresSigner && !this.signer) {
       throw new Error(
         `Cannot use ${USER_ADDRESS_PARAM} as custom parameter without a signer`
```

Comment:
Can we change this to something like the following?
```suggestion
        `Condition contains ${USER_ADDRESS_PARAM} context variable and requires a signer to populate`
```

Open to alternative wording.